From 04f96c47d13338b237eff5d314796243bc7e4a6c Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Sun, 11 Jul 2021 22:10:40 +0800
Subject: [PATCH 10/11] rockchip: disable PSCI_RESET for Pinebook Pro

Currently the mainline ATF of RK3399 only supports shutting down or
reset the whole system by writing to a GPIO passed by Coreboot. If the
system is not brought up by Coreboot, or the system have more complex
facility to power off or reboot (e.g. writing to RK808 PMIC, which is
the case of Pinebook Pro), ATF won't be able to shut down the system
correctly.

Disable PSCI_RESET for Pinebook Pro to prevent calling this problematic
codepath.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 configs/pinebook-pro-rk3399_defconfig     | 1 +
 configs/pinebook-pro-rk3399_spi_defconfig | 1 +
 2 files changed, 2 insertions(+)

diff --git a/configs/pinebook-pro-rk3399_defconfig b/configs/pinebook-pro-rk3399_defconfig
index 33b959ab47..0c0f83128f 100644
--- a/configs/pinebook-pro-rk3399_defconfig
+++ b/configs/pinebook-pro-rk3399_defconfig
@@ -10,6 +10,7 @@ CONFIG_DEBUG_UART_BASE=0xFF1A0000
 CONFIG_DEBUG_UART_CLOCK=24000000
 CONFIG_SPL_SPI_FLASH_SUPPORT=y
 CONFIG_SPL_SPI_SUPPORT=y
+# CONFIG_PSCI_RESET is not set
 CONFIG_DEBUG_UART=y
 CONFIG_BOOTDELAY=3
 CONFIG_USE_PREBOOT=y
diff --git a/configs/pinebook-pro-rk3399_spi_defconfig b/configs/pinebook-pro-rk3399_spi_defconfig
index 3cf4639927..09e5f5bb0a 100644
--- a/configs/pinebook-pro-rk3399_spi_defconfig
+++ b/configs/pinebook-pro-rk3399_spi_defconfig
@@ -13,6 +13,7 @@ CONFIG_SPL_TEXT_BASE=0xff8c2000
 # CONFIG_SPL_MMC_SUPPORT is not set
 CONFIG_SPL_SPI_FLASH_SUPPORT=y
 CONFIG_SPL_SPI_SUPPORT=y
+# CONFIG_PSCI_RESET is not set
 CONFIG_DEFAULT_DEVICE_TREE="rk3399-pinebook-pro"
 CONFIG_DEBUG_UART=y
 CONFIG_BOOTDELAY=3
-- 
2.30.2

